code/_onclick/click.dm

```diff
@@ -146,16 +146,18 @@
 
 /mob/proc/click_adjacent(atom/A, obj/item/W, mods)
 	if(W)
-		if(W.attack_speed && !src.contains(A)) //Not being worn or carried in the user's inventory somewhere, including internal storages.
-			next_move += W.attack_speed
-
-		if(!A.attackby(W, src, mods) && A && !QDELETED(A))
+		var/attackby_result = A.attackby(W, src, mods)
+		var/afterattack_result
+		if(!(attackby_result & ATTACKBY_HINT_NO_AFTERATTACK) && A && !QDELETED(A))
 			// in case the attackby slept
 			if(!W)
 				UnarmedAttack(A, 1, mods)
 				return
 
-			W.afterattack(A, src, 1, mods)
+			afterattack_result = W.afterattack(A, src, 1, mods)
+
+		if(W.attack_speed && !src.contains(A) && (attackby_result & ATTACKBY_HINT_UPDATE_NEXT_MOVE) || (afterattack_result & ATTACKBY_HINT_UPDATE_NEXT_MOVE))
+			next_move += W.attack_speed
```

Comment:
`next_move` used to be applied prior to a potential UnarmedAttack; now next_move is no longer modified if you return for an UnarmedAttack.

I have literally no idea what effects this has though; but this PR in general is a giant can of worms of possible niche interactions.

Reply:
Fixed (the UnarmedAttack next move at least, no idea about the rest of the bugs waiting to be found)